lib/navigation/service/app_router_delegates.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/authentication/view/login_view.dart';
+import 'package:acs_upb_mobile/authentication/view/sign_up_view.dart';
+import 'package:acs_upb_mobile/main.dart';
+import 'package:acs_upb_mobile/navigation/model/navigation_state.dart';
+import 'package:acs_upb_mobile/navigation/model/route_paths.dart';
+import 'package:acs_upb_mobile/navigation/view/web_shell.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+abstract class AppRouterDelegate extends RouterDelegate<RoutePath>
+    with ChangeNotifier, PopNavigatorRouterDelegateMixin<RoutePath> {
+  T _initializeProvider<T extends Listenable>(BuildContext context) {
```

Comment:
Are there any plans to use this anywhere else? Now it's only used once and I think it would be more readable if you just called `Provider.of<AuthProvider>(context)..addListener(notifyListeners)` below directly.

Reply:
I actually was, I tried passing a BuildContext to the MainRouterDelegate instead of a NavigationProvider, and use it to gather the providers I need, but doing so results in a infinite loop for some pages and I didn't find a solution up until now.